### PR TITLE
Improving ranges and index space identifiers tests

### PR DESCRIPTION
**Pull request by AmirIpma**
Solving the [issue](https://github.com/KhronosGroup/SYCL-CTS/issues/498).
Added API checks for every member function of each `item`, `nd_item` and `h_item`, error logging after device API execution.

### CODE REVIEW

File: tests/h_item/h_item_api.cpp [hunk 1]
```diff
@@ -104,46 +103,136 @@ class offset_helper {
...
+    logical_local_range,
+    physical_local_range,
+    global_id,
+    local_id,
+    logical_local_id,
+    physical_local_id,
+    methods_count
+  };
+
+  static constexpr auto method_cnt_1d = to_integral(methods_1d::methods_count);
```
[the remark this answers is not in the document]
Comment (AmirIpma) [reply tests/h_item/h_item_api.cpp:132]: Do I have to rename "methods" enums and variables to "member_functions" everywhere I use them?

**Review by ProGTX — APPROVED** [review]

**Comment by AmirIpma** [comment]
Force pushed due to incorrect commits

**Comment by bader** [comment]
@keryell, your comment has been addressed. Could you approve, please?

**Review by keryell — APPROVED** [review]
Thanks.

File: tests/nd_item/nd_item_api.cpp [hunk 2]
```diff
@@ -173,166 +230,78 @@ void populate(int *globalBuf, int *localBuf, const int *localSize,
...
-  }
-};
+TEST_CASE("sycl::nd_item<2> API", "[nd_item]") { test_item<2>(); }
 
-// construction of this proxy will register the above test
-util::test_proxy<TEST_NAME> proxy;
+TEST_CASE("sycl::nd_item<3> API", "[nd_item]") { test_item<3>(); }
 
-} /* namespace test_nd_item__ */
+} /* namespace test_nd_item__ */
```
[the remark this answers is not in the document]
Comment (AmirIpma) [reply tests/nd_item/nd_item_api.cpp]: Fixed.

**Pull request merged**